tests/clients/boto3/test_boto3_lambda.py

```diff
@@ -1,95 +1,96 @@
 # (c) Copyright IBM Corp. 2021
 # (c) Copyright Instana Inc. 2020
 
-import unittest
+import pytest
 import json
-
+from typing import Generator
 import boto3
 from moto import mock_aws
 
 from instana.singletons import tracer, agent
-from ...helpers import get_first_span_by_filter
-
-class TestLambda(unittest.TestCase):
-    def setUp(self):
-        """ Clear all spans before a test run """
-        self.recorder = tracer.recorder
+from tests.helpers import get_first_span_by_filter
+
+class TestLambda:
+    @pytest.fixture(autouse=True)
+    def _resource(self) -> Generator[None, None, None]:
+        """ Setup and Teardown """
+        # Clear all spans before a test run
+        self.recorder = tracer.span_processor
         self.recorder.clear_spans()
         self.mock = mock_aws(config={"lambda": {"use_docker": False}})
         self.mock.start()
         self.lambda_region = "us-east-1"
         self.aws_lambda = boto3.client('lambda', region_name=self.lambda_region)
         self.function_name = "myfunc"
-
-    def tearDown(self):
+        yield
         # Stop Moto after each test
         self.mock.stop()
         agent.options.allow_exit_as_root = False
 
-    def test_lambda_invoke(self):
-        with tracer.start_active_span('test'):
+    def test_lambda_invoke(self) -> None:
+        with tracer.start_as_current_span("test"):
             result = self.aws_lambda.invoke(FunctionName=self.function_name, Payload=json.dumps({"message": "success"}))
 
-        self.assertEqual(result["StatusCode"], 200)
+        assert result["StatusCode"] == 200
         result_payload = json.loads(result["Payload"].read().decode("utf-8"))
-        self.assertIn("message", result_payload)
-        self.assertEqual("success", result_payload["message"])
+        assert "message" in result_payload
+        assert "success" == result_payload["message"]
 
         spans = self.recorder.queued_spans()
-        self.assertEqual(2, len(spans))
+        assert 2 == len(spans)
 
         filter = lambda span: span.n == "sdk"
         test_span = get_first_span_by_filter(spans, filter)
-        self.assertTrue(test_span)
+        assert test_span
 
         filter = lambda span: span.n == "boto3"
         boto_span = get_first_span_by_filter(spans, filter)
-        self.assertTrue(boto_span)
+        assert boto_span
 
-        self.assertEqual(boto_span.t, test_span.t)
-        self.assertEqual(boto_span.p, test_span.s)
+        assert boto_span.t == test_span.t
+        assert boto_span.p == test_span.s
 
-        self.assertIsNone(test_span.ec)
-        self.assertIsNone(boto_span.ec)
+        assert test_span.ec is None
+        assert boto_span.ec is None
```

Comment:
```suggestion
        assert not test_span.ec
        assert not boto_span.ec
```
You can adapt the rest of the tests with this change.

Reply:
Done.